lib/slash-commands/ext/ext.js

```diff
@@ -41,31 +52,96 @@ class Ext {
+                    console.log('Could not read the Spreadsheet Configuration File.');
+                }
+                SHEET_CONFIG = data;
+                this._handle(searchString, callback);//handle the request
+            });
+        }
+
+    }
+
+    _handle(searchString, callback){
+        if (SHEET_CONFIG) {//if there's a config file we assume we should search the extension in the Spreadsheet
+            if ('update' === searchString) {//the user is asking us to update the cached extensions
+                this._request(() => callback('Lista de ramais atualizada.') );//request the spreadsheet and store its data
+            } else {
+                if (!this.dataCache.extensions) {//if there's no cache saved...
+                    this._request(() => processData(searchString, this.dataCache, callback) );//request the spreadsheet and search the extension in it
+                } else {//if already there's cached data...
+                    processData(searchString, this.dataCache, callback);//do the search for the extension
+                }
             }
+        } else {//if there's no config file we assume we have to search the extension in a file
+            util.readJsonFromFile(FILE, (err, data) => {
+                if (err) {
+                    console.error('Could not read the extensions from the File.');
+                    console.error(err);
+                    return;
+                }
+                processData(searchString, data, callback);//do the search for the extension
+            });
+        }
+    }
 
-            return callback(toString(candidates, searchString));
+    _request(callback){
+        sheet.requestSpreadsheet(SHEET_CONFIG.extensionsWorksheetId, SHEET_CONFIG.spreadsheetId, extensions => {
+            sheet.requestSpreadsheet(SHEET_CONFIG.aliasesWorksheetId, SHEET_CONFIG.spreadsheetId, aliases => {
+                this.dataCache = buildData(extensions, aliases);
+                if (callback) {
+                    callback();
+                }
+            });
         });
     }
+
 }
 
+const processData = (searchString, data, callback) => {
+    let candidates = data.extensions;
+    if (searchString) {
+        candidates = {};
+        Object.keys(data.extensions).forEach(element => {
+            if (element.match(searchString) !== null) {
+                candidates[element] = data.extensions[element];
+            }
+        });
+        // Search through aliases:
+        Object.keys(data.aliases).forEach(element => {
+            if (element.match(searchString) !== null) {
+                let key = `${ element } → ${ data.aliases[element] }`;
+                candidates[key] = data.extensions[data.aliases[element]];
+            }
+        });
+    }
+    return callback(toString(candidates, searchString));
+};
+
+const buildData = (extensions, aliases) => {
+    let data = {
+        'extensions': {},
+        'aliases': {}
+    };
+
+    for (var i in extensions) {
```

Comment:
Use `let` instead of `var`. Then you could even use `i` again in the `for` below.

Also, for `for ... in` constructs, you should use the `hasOwnProperty` check for safety and "future-proofness", otherwise it will iterate over any `super` properties as well. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in for reference.